### PR TITLE
Implement getToken callable

### PR DESCRIPTION
**Pull request by HugoGresse**
Provide a callable function to get a valid token from the client side

### CODE REVIEW

File: functions/src/db/getUserToken.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { UID } from '../types/uid'
+import { JWTToken } from '../types/JWT'
+import { db } from '../firebase/firebase'
+import { UserData } from '../types/UserData'
+
+export const getUserToken = async (userId: UID): Promise<JWTToken | null> => {
+    const userDocumentSnapshot = await db.collection('users').doc(userId).get()
+
+    if (!userDocumentSnapshot?.exists) {
+        return null
```
Comment (MattiaPrimavera) [inline functions/src/db/getUserToken.ts:10]: for every created user, a token should normally exist. If it does not, I would say it is an error case

File: functions/src/db/getUserToken.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        const validTokens = Object.keys(userData.tokens).filter(
+            (token) => userData.tokens[token].valid
+        )
+
+        if (validTokens.length > 0) {
+            return validTokens[0]
+        }
+    }
+
+    return null
```
Comment (MattiaPrimavera) [inline functions/src/db/getUserToken.ts:25]: this as well seems to me a *No valid token found* error

Comment (MattiaPrimavera) [reply functions/src/db/getUserToken.ts:25]: if this correctly returns `null`, then `getToken` as well would send back `null` to the client side, I guess

**Review by MattiaPrimavera — CHANGES_REQUESTED** [review]

**Comment by HugoGresse** [comment]
I need this for another PR, comments/review here will still be managed but in another PR.

**Pull request merged**